marlowe-cli/src/Language/Marlowe/CLI/Format.hs

```diff
@@ -0,0 +1,439 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE OverloadedStrings #-}
+
+-----------------------------------------------------------------------------
+--
+-- Module      :  Language.Marlowe.CLI.Format
+-- License     :  Apache 2.0
+--
+-- Stability   :  Experimental
+-- Portability :  Portable
+--
+-----------------------------------------------------------------------------
+
+-- | Format Marlowe contracts.
+module Language.Marlowe.CLI.Format (
+  maybeWritePretty,
+  maybeWriteJson,
+  maybeWriteYaml,
+  readContractPretty,
+  readContractJson,
+  contractParser,
+) where
+
+import Control.Monad.Combinators as C
+import Control.Monad.Except (MonadError, MonadIO, liftEither, liftIO)
+import Data.Aeson (eitherDecodeStrict)
+import Data.Bifunctor (first)
+import Data.ByteString.Base16 qualified as Base16
+import Data.ByteString.Char8 qualified as BS (getContents, pack)
+import Data.Functor (($>))
+import Data.List.NonEmpty (fromList)
+import Data.String (IsString (..))
+import Data.Text (Text)
+import Data.Text.IO qualified as T
+import Data.Void
+import GHC.Generics (Generic (..))
+import Language.Marlowe.CLI.IO (decodeFileStrict, maybeWriteJson, maybeWriteYaml)
+import Language.Marlowe.CLI.Types (CliError (..))
+import Language.Marlowe.Core.V1.Semantics.Types as S (
+  AccountId,
+  Action (..),
+  Bound (..),
+  Case (..),
+  ChoiceId (..),
+  Contract (..),
+  CurrencySymbol (..),
+  Observation (..),
+  Party (..),
+  Payee (..),
+  Timeout,
+  Token (..),
+  TokenName (..),
+  Value (..),
+  ValueId (..),
+ )
+import Language.Marlowe.Core.V1.Semantics.Types.Address
+import Language.Marlowe.Pretty
+import Plutus.V1.Ledger.Time (POSIXTime (..))
+import PlutusTx.Builtins (BuiltinByteString)
+import PlutusTx.Builtins.Class (ToBuiltin (..))
+import Text.Megaparsec
+import Text.Megaparsec.Char as P
+import Text.Megaparsec.Char.Lexer qualified as L
+
+-- | Read a Marlowe Contract as JSON from a file.
+readContractJson
+  :: (MonadError CliError m)
+  => (MonadIO m)
+  => Maybe FilePath
+  -> m Contract
+readContractJson (Just inputFile) =
+  decodeFileStrict inputFile
+readContractJson Nothing =
+  (liftEither . first (CliError . show)) . eitherDecodeStrict
```

Comment:
This function is called for both JSON and YAML input. Does `Data.Aeson.eitherDecodeStrict` handle YAML in addition to JSON, or do we need to use `Data.Yaml.decodeEither`?